### PR TITLE
[13.0][ADD] l10n_th_amount_to_text

### PR DESCRIPTION
**Pull request by Saran440**
Converted Amount Text from english to thai by send context lang = 'th_TH' to res.currency
You can applied this module to base of every form that convert amount to text thai

- Normally, Core odoo has function amount_to_text and return text language from context or user setting
Example : 45.75 Baht
user language 'Thai' -> สี่สิบห้า Baht และ เจ็ดสิบห้า Satang
user language 'English' -> Forty-Five Baht and Seventy-Five Satang

- This module will change check context lang. if you send context lang = 'th_TH',
it will convert to สี่สิบห้าบาทเจ็ดสิบห้าสตางค์

- Supported multi-currency (EUR and USD).
if you send context lang thai but amount is 45.75 Euros,
it will convert to สี่สิบห้ายูโรเจ็ดสิบห้าเซนต์

**NOTE** : In core odoo required package num2words version 0.5.6 but amount thai text need version 0.5.7+ Please upgrade package before that)

### CODE REVIEW

File: l10n_th_amount_to_text/models/res_currency.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+_logger = logging.getLogger(__name__)
+
+try:
+    from num2words import num2words
+except ImportError:
+    _logger.warning(
+        "The num2words python library is not installed,"
+        "amount-to-text features won't be fully available."
+    )
+    num2words = None
```
Comment (ps-tubtim) [inline l10n_th_amount_to_text/models/res_currency.py]: This code is the same as the odoo core.
https://github.com/odoo/odoo/blob/48f3fa1673cabaa6b4031fbe524be632d3897ec0/odoo/addons/base/models/res_currency.py#L14

**Review by ps-tubtim — COMMENTED** [review]
Function Test: Ok.

**Comment by Saran440** [comment]
@ps-tubtim Fixed and enhance code for support multi-currency.
Thank you for your reviewed. I will add test-script later.

File: l10n_th_amount_to_text/tests/test_amount_to_text.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class TestAmountToText(SavepointCase):
+    def test_currency_th_amount_to_text(self):
+        """ verify that amount_to_text converted text to thai language """
+        currency = self.env.ref("base.THB")
+        amount_text_en = currency.amount_to_text(1050.75)
+        self.assertEqual(
+            amount_text_en, "One Thousand And Fifty Baht and Seventy-Five Satang"
+        )
+        currency.with_context({"lang": "th_TH"}).amount_to_text(1050.75)
+        # num2words need version 0.5.7+
```
Comment (kittiu) [inline l10n_th_amount_to_text/tests/test_amount_to_text.py]: You need these comment? or await to write more?

File: l10n_th_amount_to_text/tests/test_amount_to_text.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        # amount_text_th = currency.with_context({"lang": "th_TH"}).amount_to_text(
+        #     1050.75
+        # )
+        # self.assertEqual(amount_text_th, "หนึ่งพันห้าสิบบาทเจ็ดสิบห้าสตางค์")
```
Comment (kittiu) [inline l10n_th_amount_to_text/tests/test_amount_to_text.py]: You comment out test?

Comment (Saran440) [reply l10n_th_amount_to_text/tests/test_amount_to_text.py]: core odoo used num2words v.0.5.6 (exclude th currency) but this module need num2words v.0.5.7 that I can't test script this function

**Review by ps-tubtim — APPROVED** [review]
Code and Functional Test: OK.

**Review by kittiu — APPROVED** [review]

**Pull request merged**